### PR TITLE
Update uniffi dependency to v0.9.0

### PR DESCRIPTION
**Pull request by rfk**
There was a bit of churn here in the `fxa-client` component to make it `[Threadsafe]` and avoid the new deprecation warning, but otherwise this seems straightforward. Pinging both Sync and Nimbus folks for review from a risk-mitigation perspective, given any upcoming releases.

### CODE REVIEW

File: components/fxa-client/src/lib.rs [hunk 1]
```diff
@@ -104,7 +104,9 @@ pub enum FxaError {
 /// account and accessing other services on behalf of the user.
 ///
 pub struct FirefoxAccount {
-    internal: internal::FirefoxAccount,
+    // For now, we serialize all access on a single `Mutex` for thread safety across
+    // the FFI. We should make the locking more granular in future.
+    internal: std::sync::Mutex<internal::FirefoxAccount>,
```
Comment (rfk) [inline components/fxa-client/src/lib.rs:109]: Right now I am very happy that I went with this `internal` abstraction that could be easily wrapped in a single Mutex 🎉

Comment (rfk) [reply components/fxa-client/src/lib.rs:109]: (For context: the previous version of UniFFI was adding a Mutex like this automagically, but that behaviour is deprecated and will be removed in a future version)

**Comment by rfk** [comment]
This is currently failing CI on an Android test issue, but I think that may have been introduced by https://github.com/mozilla/application-services/pull/4129.

**Review by mhammond — APPROVED** [review]
🚀 🚢

**Comment by rfk** [comment]
Since we just cut a new release for Nimbus, I'm going to go ahead and merge this once CI is green.

**Pull request merged**